File: Java Solutions/Math/palindromeNumber.java
//LeetCode: Palindrome Number
//Determine whether an integer is a palindrome. Do this without extra space.

public class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;
        int div = 1;
        while (x / div >= 10) {
            div *= 10; 
        }
        while (x != 0) {
            int lastDigit = x % 10;
            int leftPart = x / div;
            if (leftPart != lastDigit) return false;
            x = (x % div) / 10;
            //cut the head and the tail
            div /= 100;
        }
        return true;
    }
}

File: C++ Solutions/Math & Bit Manipulation/PalindromeNumber.cpp
//Palindrome Number
/*Determine whether an integer is a palindrome. Do this without extra space.

click to show spoilers.

Subscribe to see which companies asked this question.
*/

class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0) return false;
        if(x == 0) return true;
        int div = 1;
        while(x/div >= 10){
            div *= 10;
        }
        while(x != 0){
            int l = x/div;
            int r = x%10;
            if(l != r) return false;
            x = (x%div)/10;
            div = div/100;
        }
        return true;
    }
};
